### ISSUE TITLE
continueWithRedirectUri does not continue Plaid Link flow 

### ISSUE THREAD

**Issue by alefischer13** [jorgefspereira/plaid_flutter#74]
Hello,

I'm using `PlaidLink.continueWithRedirectUri` when testing OAuth app-to-app. After finishing the connect flow in the 3rd-party bank app, my application opens, I see "Continue with sign in" on the Plaid Link for a split second, and then it closes and sends me back to the page where I initiate `PlaidLink.open()`. I ensured that the redirect uri is the one we are expecting (oauth_state_id=<xyz>), and other non-app-to-app flows work fine. I tried with both versions 2.2.1 and 2.2.2 and I get the same error. 
Your help would be sincerely appreciated!

Alejandro

**Comment by yaroslav-kahaniak** [comment]
Hello,

I am experiencing the same issue. `PlaidLink.onSuccess` is never called. Here is the latest event that app receives from the `PlaidLink.onEvent` stream after calling `PlaidLink.continueWithRedirectUri`:
EXIT: {"viewName":"","exitStatus":"","mfaType":"","requestId":"ccjh9NuzQxQLFcw","timestamp":"2023-03-09T16:01:29Z","linkSessionId":"","institutionName":"","institutionId":"","institutionSearchQuery":"","errorType":"INVALID_REQUEST","errorCode":"INVALID_FIELD","errorMesssage":"invalid continuation token"}

**Package version:** 3.1.0
**flutter --version:**
Flutter 3.3.10
Framework • revision 135454af32 (3 months ago) • 2022-12-15 07:36:55 -0800
Engine • revision 3316dd8728
Tools • Dart 2.18.6 • DevTools 2.15.0

@jorgefspereira
I would be thankful for any help

**Comment by yaroslav-kahaniak** [comment]
Hello @jorgefspereira  ,
Could you tell, please, if you have time/intention to dedicate some work to this bug in the nearest future?

**Comment by jorgefspereira** [comment]
@yaroslav-kahaniak I'll try to reproduce the error on my test example still this week.

**Comment by jorgefspereira** [comment]
@yaroslav-kahaniak can you give more the details of your test case (platform, token type, etc)?

**Comment by yaroslav-kahaniak** [comment]
@jorgefspereira , sure
Platform: iOS 16.2
Link configuration: LinkTokenConfiguration

Backend passes these params into the **POST /link/token/create** (https://plaid.com/docs/api/tokens/#linktokencreate) request:
```
{
   "products": ["transactions"],
   "country_codes": ["US"],
   "language": "en",
}

```
**Mentioned earlier:**
Package version: 3.1.0
flutter --version:
Flutter 3.3.10
Framework • revision 135454af32 (3 months ago) • 2022-12-15 07:36:55 -0800
Engine • revision 3316dd8728
Tools • Dart 2.18.6 • DevTools 2.15.0

**Comment by nzkn** [comment]
@jorgefspereira I wonder, are you planning to work on this issue any time soon? It is stopping the app I am working on from release. So if you are not going to work on that soon, I can take a closer look into the package source code and try to resolve it this weekend

**Comment by jorgefspereira** [comment]
@nzkn I was testing it right now. Can you give more info of your token creation process and the out put error? Fell free to take a closer look to the source code and try to fix the issue.

**Comment by nzkn** [comment]
The plaid link is generated on the backed.
The following plaid config is used in node server:
```
plaid: {
    clientId: clientId,
    secret: secret,
    env: 'sandbox',
    version: '2020-09-14',
    tokenOptions: {
      client_name: clientName,
      products: ['auth'],
      country_codes: ['US'],
      language: 'en',
      redirect_uri: redirectUrl,
    }
  },
```

The server makes request to PlaidApi and returns link_token like this:
```
async getToken() {
  const plaidClient = new PlaidApi(configuration);
  const response = await plaidClient.linkTokenCreate({
    user: {
      client_user_id: req.user.id,
    },
    ...plaidConfig.tokenOptions,
  });
  return {linkToken: respoinse.data.link_token}
}
```

Before retrieving linkToken listeners are added in the app:
```
_plaidExit = PlaidLink.onExit.listen...
_plaidSuccess = PlaidLink.onSuccess.listen...
```

API call is made to the server, and plaidLink returned to launch Plaid:
```
var plaidLink = await getPlaidIntegrationLink();
var configuration = LinkTokenConfiguration(token: plaidLink.linkToken);
await PlaidLink.open(configuration: configuration);
```

After the sheet is opened, I go step-by-step and enable integration with Chase.

Neither `PlaidLink.onSuccess` nor `PlaidLink.onExit` gets called.

**Comment by jorgefspereira** [comment]
Hi @nzkn, 

To test this i created a simple firebase project to host my apple-app-site-association file.

I added my domain to the allowed redirect URIs:

<img width="388" alt="Screenshot 2023-04-03 at 11 13 18" src="https://user-images.githubusercontent.com/11697952/229481659-b2a105a5-d11f-4adb-b7ce-4c63a26d0232.png">

I created the link token with the following parameters:

<img width="385" alt="Screenshot 2023-04-03 at 11 12 27" src="https://user-images.githubusercontent.com/11697952/229481857-c9454f1a-0eb1-48d6-b512-52092cb1002b.png">

And ran the provided example of this plugin.

![Simulator Screen Recording - iPhone 14 Pro - 2023-04-03 at 11 24 17](https://user-images.githubusercontent.com/11697952/229483738-5de7fe0a-ca8a-48e2-b92d-b6fe51b37d7f.gif)

The flow worked ok, and I also received the onSuccess messages:

<img width="372" alt="Screenshot 2023-04-03 at 11 24 23" src="https://user-images.githubusercontent.com/11697952/229484051-a2149bf6-20a8-4663-bdd1-438107a49ad2.png">

Is there something I'm not taking into account? I'll also test today the continueWithRedirectUri function to be sure if something is wrong

**Comment by nzkn** [comment]
@jorgefspereira Works fine for me too when I am not calling `continueWithRedirectUri`. I wonder, am I doing something wrong by calling `PlaidLink.continueWithRedirectUri` with the deep link that opens the app? Should I even call this function?

**Comment by jorgefspereira** [comment]
@nzkn no, you're right. `continueWithRedirectUri` should be call when your app receives a universal link. It should handle the support for app-to-app oauth. Let me try to check this app-to-app flow to see if something is wrong with it.

**Comment by jagloro** [comment]
@jorgefspereira I put together a private repo with an example [here](https://github.com/jagloro/flutter_plaid_demo) and added you as a collaborator. 

Could you take a look?

**Comment by jorgefspereira** [comment]
> @jorgefspereira Thank you for the update! Let's wait for Plaid's response. Could you share the progress of your Plaid's investigation into this thread once there is any?

Sure I'll add here as soon as they have some feedback.
